### PR TITLE
Allow to load order, invoice, shipment and creditmemo by increment id

### PR DESCRIPTION
**Pull request by luigifab**
### Description

This PR allow to load order, invoice, shipment and creditmemo by increment id from URL.

Go to _Sales / Orders_, or _Sales / Invoices_, or _Sales / Shipments_, or _Sales / Credit memos_.
Open any item.
Your URL looks like: `.../index.php/admin/admin/sales_order/view/order_id/2/...`
Now replace the order_id by the increment_id, for example: `.../index.php/admin/admin/sales_order/view/order_id/100000002/...`

And, it works!

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

File: app/code/core/Mage/Sales/Model/Order/Creditmemo.php [hunk 1]
```diff
@@ -224,6 +224,25 @@ protected function _initOldFieldsMap()
...
+    public function loadByIncrementId($incrementId)
+    {
+        $ids = $this->getCollection()
+            ->addAttributeToFilter('increment_id', $incrementId)
+            ->getAllIds();
+
+        if (!empty($ids)) {
+            reset($ids);
+            $this->load(current($ids));
+        }
```
Comment (kiatng) [inline app/code/core/Mage/Sales/Model/Order/Creditmemo.php:242]:
```suggestion
        return $this->load($incrementId, 'increment_id');
```

Comment (luigifab) [reply app/code/core/Mage/Sales/Model/Order/Creditmemo.php:242]: I copied/pasted _Mage_Sales_Model_Order_Invoice_, but why not!

**Comment by sreichel** [comment]
> Now replace the order_id by the increment_id, for example: .../index.php/admin/admin/sales_order/view/order_id/100000002/...

Also change the parameter to `increment_id`?

**Comment by luigifab** [comment]
For #2726 I can't.

**Comment by sreichel** [comment]
When the url paramenter is called order_id and dont expect to pass different values to it.

**Comment by luigifab** [comment]
Yes, I did that for customer support.

When customers contact us, they use **increment_id**, so our customer agents go to _Sales / Order_, then fill the _increment_id_, then press enter or filter button, then **wait**, then click on the order to open it. This is time consuming.

With this change, they can keep one tab with any order, and update the `order_id` parameter with any `increment_id`. Some agents prefer a search bar, I also added it.

Because this is a simple change, I think/hope that it will don't break anything, so I share it.

Yes it's better for dev people to use `order_id|invoice_id|shipment_id|creditmemo_id` parameter (`entity_id` in `sales_flat_[order|invoice|shipment|creditmemo]_grid`) and another `increment_id` parameter (`increment_id` in `sales_flat_[order|invoice|shipment|creditmemo]_grid`); but for users, it's easier to only update the parameter value.

**Review by kiatng — APPROVED** [review]
It's a useful hack.

**Comment by fballiano** [comment]
I don't know about this, I see the customer service point but I kinda not like the idea of loading something by some attribute when clearly the parameter name shows something else...

**Pull request closed without merging**